cmake/cmakelibs.cmake

```diff
@@ -0,0 +1,180 @@
+    set(OSQUERY_EXTENSION_GROUP_VERSION "1.0")
+  endif()
+
+  # Build the include list; this contains the files required to declare
+  # the classes used in the REGISTER_EXTERNAL directives
+  # Note: The variables in uppercase are used by the template
+  get_property(main_include_list GLOBAL PROPERTY OSQUERY_EXTENSION_GROUP_MAIN_INCLUDES)
+  foreach(include_file ${main_include_list})
+    set(OSQUERY_EXTENSION_GROUP_INCLUDES "${OSQUERY_EXTENSION_GROUP_INCLUDES}\n#include <${include_file}>")
+  endforeach()
+
+  # We need to generate the main.cpp file, containing all the required
+  # REGISTER_EXTERNAL directives
+  get_property(OSQUERY_EXTENSION_GROUP_INITIALIZERS GLOBAL PROPERTY OSQUERY_EXTENSION_GROUP_INITIALIZERS)
+  configure_file(
+    "${CMAKE_SOURCE_DIR}/tools/codegen/templates/osquery_extension_group_main.cpp.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/osquery_extension_group_main.cpp"
+  )
+
+  # Extensions can no longer control which compilation flags to use here (as they are shared) so
+  # we are going to enforce sane defaults
+  if(DEFINED PLATFORM_POSIX)
+    set(extension_cxx_flags ${cxx_settings} -DBOOST_ASIO_DISABLE_STD_STRING_VIEW)
+  else()
+    set(extension_cxx_flags /W4)
+  endif()
+
+  # Generate the extension target
+  add_executable("${OSQUERY_EXTENSION_GROUP_NAME}"
+    "${CMAKE_CURRENT_BINARY_DIR}/osquery_extension_group_main.cpp"
+    ${extension_source_files}
+  )
+
+  set_property(TARGET "${OSQUERY_EXTENSION_GROUP_NAME}" PROPERTY INCLUDE_DIRECTORIES "")
+  target_compile_options("${OSQUERY_EXTENSION_GROUP_NAME}" PRIVATE ${extension_cxx_flags})
+
+  target_link_libraries("${OSQUERY_EXTENSION_GROUP_NAME}" PUBLIC
+  	osquery_sdk_pluginsdk
+  	osquery_extensions_implthrift
+	thirdparty_boost
+  )
+  
+  if(DEFINED PLATFORM_LINUX)
+    target_link_libraries("${OSQUERY_EXTENSION_GROUP_NAME}" PUBLIC
+	  thirdparty_libiptc
+  	)
+  endif()
+
+  set_target_properties("${OSQUERY_EXTENSION_GROUP_NAME}" PROPERTIES
+    OUTPUT_NAME "${OSQUERY_EXTENSION_GROUP_NAME}.ext"
+  )
+
+  # Import the core libraries; note that we are going to inherit include directories
+  # with the wrong scope, so we'll have to fix it
+  set_property(TARGET "${OSQUERY_EXTENSION_GROUP_NAME}" PROPERTY INCLUDE_DIRECTORIES "")
+  target_include_directories("${OSQUERY_EXTENSION_GROUP_NAME}" SYSTEM PRIVATE 
+    ${include_folder_list}
+  )
+
+  # Apply the user (extension) settings
+  get_property(library_list GLOBAL PROPERTY OSQUERY_EXTENSION_GROUP_LIBRARIES)
+  if(NOT "${library_list}" STREQUAL "")
+    target_link_libraries("${OSQUERY_EXTENSION_GROUP_NAME}" PUBLIC ${library_list})
+  endif()
+
+  get_property(include_folder_list GLOBAL PROPERTY OSQUERY_EXTENSION_GROUP_INCLUDE_FOLDERS)
+  if(NOT "${include_folder_list}" STREQUAL "")
+    target_include_directories("${OSQUERY_EXTENSION_GROUP_NAME}" PRIVATE
+      ${include_folder_list}
+    )
+  endif()
+endfunction()
+
+function(add_osquery_extension_ex class_name extension_type extension_name ${ARGN})
```

Comment:
This is an interface function, so we remove it completely; we could however redirect it to addOsqueryExtensionEx and print a "the add_osquery_extension_ex function is deprecated, please use addOsqueryExtensionEx" message when it is called